src/server.c

```diff
@@ -894,7 +894,7 @@ void updateClientMemoryUsage(client *c) {
 }
 
 int clientEvictionAllowed(client *c) {
-    if (server.maxmemory_clients == 0 || c->flag.no_evict || !c->conn) {
+    if (server.maxmemory_clients == 0 || c->flag.no_evict || c->flag.fake) {
         return 0;
     }
     int type = getClientType(c);
```

Comment:
serverAssert on `c->conn`?

Reply:
This is good practice. But how do we ensure a new dev/reviewer ensure(s) this? Wouldn't this be better asserted while setting/unsetting the flag in one place? WDYT?

Reply:
> Wouldn't this be better asserted while setting/unsetting the flag in one place?

I think we would need to create a wrapper like the below and make `flag.fake` a `client` internal state.

```c
connection *clientGetConnection(client *c) {
    serverAssert((c->flag.fake && !c->conn) || (!c->flag.fake && c->conn));
    return c->conn;
}
```

Reply:
the wrapper changed will require a lot of code changes, there a LOT OF code are using c->conn

Reply:
@enjoy-binbin We merge this and file a new issue with `help-wanted`/`good-first-issue` tag?

Reply:
I'm not worried about not doing it myself, i just wonder if it's worth doing (diff causes conflicts).

But looking at the code again, i guess doing this wrapper is indeed a good way to avoid fake client connection issue in the future. I will do it in a later commit.

Reply:
I'm starting to feel like we might still need to keep c->conn...